### ISSUE TITLE
Exception on /api/1/channels [PUT]

### ISSUE THREAD

**Issue by palango** [raiden-network/raiden#1820]
## Problem Definition

When opening a a new channel from the WebUI:

```
2018-07-13 15:20:32 [error    ] Exception on /api/1/channels [PUT] [flask.app]
Traceback (most recent call last):
  File "/Users/paul/Work/venv-raiden/lib/python3.6/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/paul/Work/venv-raiden/lib/python3.6/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/paul/Work/venv-raiden/lib/python3.6/site-packages/flask_restful/__init__.py", line 480, in wrapper
    resp = resource(*args, **kwargs)
  File "/Users/paul/Work/venv-raiden/lib/python3.6/site-packages/flask/views.py", line 88, in view
    return self.dispatch_request(*args, **kwargs)
  File "/Users/paul/Work/venv-raiden/lib/python3.6/site-packages/flask_restful/__init__.py", line 595, in dispatch_request
    resp = meth(*args, **kwargs)
  File "/Users/paul/Work/venv-raiden/lib/python3.6/site-packages/webargs/core.py", line 441, in wrapper
    return func(*args, **kwargs)
  File "/Users/paul/Work/raiden/raiden/api/v1/resources.py", line 53, in put
    **kwargs,
  File "/Users/paul/Work/raiden/raiden/api/rest.py", line 383, in open
    reveal_timeout,
  File "/Users/paul/Work/raiden/raiden/api/python.py", line 237, in channel_open
    retry_timeout,
  File "/Users/paul/Work/raiden/raiden/waiting.py", line 53, in wait_for_newchannel
    gevent.sleep(retry_timeout)
  File "/Users/paul/Work/venv-raiden/lib/python3.6/site-packages/gevent/hub.py", line 155, in sleep
    hub.wait(t)
  File "src/gevent/_hub_primitives.py", line 46, in gevent.__hub_primitives.WaitOperationsGreenlet.wait
  File "src/gevent/_hub_primitives.py", line 55, in gevent.__hub_primitives.WaitOperationsGreenlet.wait
  File "src/gevent/_waiter.py", line 152, in gevent.__waiter.Waiter.get
  File "src/gevent/_greenlet_primitives.py", line 59, in gevent.__greenlet_primitives.SwitchOutGreenletWithLoop.switch
  File "src/gevent/_greenlet_primitives.py", line 59, in gevent.__greenlet_primitives.SwitchOutGreenletWithLoop.switch
  File "src/gevent/_greenlet_primitives.py", line 63, in gevent.__greenlet_primitives.SwitchOutGreenletWithLoop.switch
  File "src/gevent/__greenlet_primitives.pxd", line 35, in gevent.__greenlet_primitives._greenlet_switch
raiden.exceptions.EthNodeCommunicationError: After 180 seconds the channel was not properly created.
```

**Comment by LefterisJP** [comment]
I would not call this a bug. 

We have had this discussion in the past. We need to decide how to handle calls that wait on a blockchain transaction to get mined. For the moment we have the configurable polling timeout which is 180 seconds.

Depending on user configuration, gas prices and congestion a transaction may indeed not get mined at all or take a very very long time.

The choices we have is to:
1. Either block indefinitely until a transaction gets mined 
2. Or do what we do now and have a polling timeout after which we just give up. The local state **should** anyway get updated after the event is seen in the blockchain.

**Comment by LefterisJP** [comment]
> we should catch these kind of exceptions and log a more meaningful message.

That for sure. I wanted to do that in a PR soon.

**Comment by ulope** [comment]
I’d say the real solution is to switch to an async API.

**Comment by LefterisJP** [comment]
> I’d say the real solution is to switch to an async API.

That also would need discussion (and is out of scope for this release as it would be too big a change), but I can see the merits of an async API approach with anything having to do with the chain.

**Comment by palango** [comment]
My proposal would be to catch the exception and return a [HTTP 202](https://httpstatuses.com/202) instead when the timeout is hit. Opinions?

**Comment by hackaugusto** [comment]
> My proposal would be to catch the exception and return a HTTP 202 instead when the timeout is hit. Opinions?

It looks like a good response for this.

> The choices we have is to:
> Either block indefinitely until a transaction gets mined
> Or do what we do now and have a polling timeout after which we just give up. The local state should anyway get updated after the event is seen in the blockchain.

I think we should either block until the transaction leaves the transaction poll (either because it was mined or because it was overwritten), or default to async apis that return `202` right away.

I still think that having these timeouts on the Raiden client is a really bad idea, because timeouts don't compose well, and we cant really choose a good value, so that decision should be pushed to the client.
